flaml/data.py

```diff
@@ -381,10 +381,23 @@ def fit_transform(self, X: Union[DataFrame, np.array], y, task):
             y = self.label_transformer.fit_transform(y)
         else:
             self.label_transformer = None
+            if task == TOKENCLASSIFICATION:
+                # if the labels are tokens, convert them to ids
+                if any([isinstance(x, str) for x in y[0]]):
+                    self._label_list = sorted(list(set().union(*y)))
+                    self._tokenlabel_to_id = {
+                        self._label_list[x]: x for x in range(len(self._label_list))
+                    }
+                    y = y.apply(lambda x: [self._tokenlabel_to_id[xx] for xx in x])
+                # if the labels are not tokens, they must be ids
+                else:
+                    assert all(
+                        [isinstance(x, int) for x in y[0]]
+                    ), "The labels must either be tokens or ids"
         self._task = task
         return X, y
 
-    def transform(self, X: Union[DataFrame, np.array]):
+    def transform(self, X: Union[DataFrame, np.array], y=None):
```

Comment:
We need to consolidate the transformation of `y` in one place only. Right now that transformation is done sometimes by `self.label_transformer` and sometimes by `transform()`. That's not good.